internal/commands/pull/inputs_test.go

```diff
@@ -116,6 +132,19 @@ func TestPullInputsResolveRemoteApp(t *testing.T) {
 		assert.Equal(t, realm.AppFilter{GroupID: "some-project", App: "some-app"}, appFilter)
 	})
 
+	t.Run("should not resolve group id if app meta exists", func(t *testing.T) {
+		i := inputs{appMeta: local.AppMeta{GroupID: "some-group", AppID: "some-app", ConfigVersion: realm.DefaultAppConfigVersion}}
+		var realmClient mock.RealmClient
+
+		realmClient.FindAppFn = func(groupID, appID string) (realm.App, error) {
+			return realm.App{GroupID: "group-id", ID: "app-id", Name: "some-name"}, nil
+		}
+
+		app, err := i.resolveRemoteApp(nil, cli.Clients{Realm: realmClient})
+		assert.Nil(t, err)
+		assert.Equal(t, realm.App{GroupID: "group-id", ID: "app-id", Name: "some-name"}, app)
```

Comment:
[q] I might be missing sth here but shouldn't the `GroupID = "some-group"` (i.e the one provided by the appMeta)?
aren't the app details being set with the app meta here: https://github.com/makesitgo/realm-cli/blob/2e0c742f99998e75f27bf91e55047d1a07550158/internal/cli/inputs.go#L86

Reply:
good callout... there's a funny `TODO` in the `pull_inputs.resolveRemoteApp` function which, after resolving the app, will specifically use `FindApp` to fetch fuller app details (in particular, the "template id").

tweaked this to have the mock "pass through" app data and assert that the `FindApp` function is only called once, rather than twice